tests/robustness/traffic/key_window.go

```diff
@@ -0,0 +1,62 @@
+package traffic
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+)
+
+type KeyWindow struct {
+	sync.Mutex
+
+	Counter    int
+	WindowSize int
+	Keys       []int
+	keyPrefix  string
+}
+
+func NewKeyWindow(size int, keyPrefix string) *KeyWindow {
+	k := &KeyWindow{
+		Counter:    0,
+		WindowSize: size,
+		Keys:       make([]int, size),
+		keyPrefix:  keyPrefix,
+	}
+
+	for k.Counter < size {
+		k.Keys[k.Counter] = k.Counter
+		k.Counter++
+	}
+
+	return k
+}
+
+func (k *KeyWindow) GetKey() string {
+	k.Lock()
+	defer k.Unlock()
+
```

Comment:
I think we can structure this function by making it like this:
```go
key := rand.Intn(k.WindowSize)
if k.useWindow() {
  randkey := key
  key = k.Keys[randKey]
  k.Keys[randKey] = k.Counter
  k.Counter++
}
return fmt.Sprintf("%s%d", k.keyPrefix, key)
```

Reply:
no hold on one sec on this one.

Reply:
nvm, I think I kinda got it right, confused myself about WindowSize and the random value in useWindow().